client/router.js

```diff
@@ -120,20 +120,24 @@ const options = {
 export default function getRouter() {
   const router = new Router(options);
   router.beforeEach((to, from, next) => {
-    const auth = router.app.$store.state.auth;
-    if (isAuthRequired(to, auth)) {
-      return next({ path: '/login', query: { redirect: to.fullPath } });
-    }
-    return isAllowed(to, auth) ? next() : next({ path: from.fullPath });
+    const { user } = router.app.$store.state.auth;
+    if (isAuthRequired(to, user)) return redirectToLogin(to, next);
```

Comment:
This might be a bit clearer if isAuthRequired would just check is auth required and not check if a user is logged in 😉  💅